cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -475,27 +476,64 @@ protected Future<Map<String, Object>> createOrUpdateConnector(Reconciliation rec
         return apiClient.createOrUpdatePutRequest(reconciliation, host, port, connectorName, asJson(connectorSpec, desiredConfig))
             .compose(ignored -> apiClient.statusWithBackOff(reconciliation, new BackOff(200L, 2, 10), host, port,
                     connectorName))
-            .compose(status -> pauseResume(reconciliation, host, apiClient, connectorName, connectorSpec, status))
+            .compose(status -> updateState(reconciliation, host, apiClient, connectorName, connectorSpec, status, new ArrayList<>()))
             .compose(ignored ->  apiClient.status(reconciliation, host, port, connectorName));
     }
 
-    private Future<Void> pauseResume(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, KafkaConnectorSpec connectorSpec, Map<String, Object> status) {
+    private Future<List<Condition>> updateState(Reconciliation reconciliation, String host, KafkaConnectApi apiClient, String connectorName, KafkaConnectorSpec connectorSpec, Map<String, Object> status, List<Condition> conditions) {
         @SuppressWarnings({ "rawtypes" })
         Object path = ((Map) status.getOrDefault("connector", emptyMap())).get("state");
-        if (!(path instanceof String)) {
+        if (!(path instanceof String state)) {
             return Future.failedFuture("JSON response lacked $.connector.state");
         } else {
-            String state = (String) path;
-            boolean shouldPause = Boolean.TRUE.equals(connectorSpec.getPause());
-            if ("RUNNING".equals(state) && shouldPause) {
-                LOGGER.debugCr(reconciliation, "Pausing connector {}", connectorName);
-                return apiClient.pause(reconciliation, host, port, connectorName);
-            } else if ("PAUSED".equals(state) && !shouldPause) {
-                LOGGER.debugCr(reconciliation, "Resuming connector {}", connectorName);
-                return apiClient.resume(reconciliation, host, port, connectorName);
-            } else {
-                return Future.succeededFuture();
+            ConnectorState desiredState = connectorSpec.getState();
+            @SuppressWarnings("deprecation")
+            Boolean shouldPause = connectorSpec.getPause();
+            ConnectorState effectiveState = desiredState != null ? desiredState :
+                    Boolean.TRUE.equals(shouldPause) ? ConnectorState.PAUSED : ConnectorState.RUNNING;
+            if (desiredState != null && shouldPause != null) {
+                String message = "Both pause and state are set. Since pause is deprecated, state takes precedence " +
```

Comment:
Should I use `$.connector.` like in the message above, or is this fine?

Reply:
No, if you want, you can do `.spec.state` and `.spec.pause`. But not the `$`.

Reply:
Ok, I'll leave it as is